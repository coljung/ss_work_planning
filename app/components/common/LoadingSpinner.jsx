import React from 'react';
import { Spin } from 'antd';
import PropTypes from 'prop-types';

const LoadingSpinner = ({ text, classUsed = 'spinDiv' }) => {
    const loadingText = text || 'Loading...';

    return (
        <div className={classUsed}>
            <Spin size="large" style={{ fontSize: 20, color: '#777' }} tip={loadingText}/>
        </div>
    );
};

LoadingSpinner.propTypes = {
    text: PropTypes.string,
    classUsed: PropTypes.string,
};

export default LoadingSpinner;
