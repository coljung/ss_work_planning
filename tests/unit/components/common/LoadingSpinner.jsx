import React from 'react';
import { shallow } from 'enzyme';
import { Spin } from 'antd';
import LoadingSpinner from '../../../../app/components/common/LoadingSpinner';

describe('<LoadingSpinner />', () => {
    it('should render base layout', () => {
        const wrapper = shallow(
            <LoadingSpinner />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('should render base layout with default text', () => {
        const wrapper = shallow(
            <LoadingSpinner />
        );

        const spin = wrapper.find(Spin).first();
        expect(spin.prop('tip')).toEqual('Loading...');
    });

    it('should render base layout with custom text', () => {
        const wrapper = shallow(
            <LoadingSpinner text="Test" />
        );

        const spin = wrapper.find(Spin).first();
        expect(spin.prop('tip')).toEqual('Test');
    });
});
