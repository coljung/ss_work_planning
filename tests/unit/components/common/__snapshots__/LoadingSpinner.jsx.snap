// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> should render base layout 1`] = `
<Spin
  prefixCls="ant-spin"
  size="large"
  spinning={true}
  tip="Loading..."
  wrapperClassName=""
/>
`;
